app/src/main/java/com/gh4a/service/RepositoryService.java

```diff
@@ -0,0 +1,153 @@
+package com.gh4a.service;
+
+import android.app.Activity;
+import android.util.Log;
+import com.gh4a.DefaultClient;
+import com.gh4a.Gh4Application;
+import com.gh4a.utils.HtmlUtils;
+import com.gh4a.utils.rx.RxTools;
+import org.eclipse.egit.github.core.Repository;
+import org.eclipse.egit.github.core.RepositoryId;
+import org.eclipse.egit.github.core.client.GitHubClient;
+import org.eclipse.egit.github.core.client.RequestException;
+import org.eclipse.egit.github.core.service.ContentsService;
+import org.eclipse.egit.github.core.service.IssueService;
+import org.eclipse.egit.github.core.service.StarService;
+import org.eclipse.egit.github.core.service.WatcherService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Locale;
+import io.reactivex.Observable;
+
+public class RepositoryService {
+    public static final int LOAD_README = 0;
+    public static final int LOAD_PULL_REQUESTS_COUNT = 1;
+    public static final int UPDATE_STAR = 2;
+    public static final int UPDATE_WATCH = 3;
+    public static final int IS_STARRING = 4;
+    public static final int IS_WATCHING = 5;
+    public static final int LOAD_REPOSITORY = 6;
+
+    public static Observable loadReadme(Activity activity, String repoOwner, String repoName, String ref) {
+        Gh4Application app = (Gh4Application) activity.getApplicationContext();
+
+        return Observable.create(emitter -> {
```

Comment:
Use Single instead of Observable as there is just a single result value.

Reply:
Well noticed, It would be more appropriate ... but unfortunately the following [issue](https://github.com/kmdupr33/RxLoader/issues/24) on  `SingleTransformer` method exposure does not allow me transforming `Single` streams yet